src/modules/blogs/blog.controller.ts

```diff
@@ -121,4 +122,145 @@ export class BlogController {
       res.status(500).send({ message: "Internal server error" });
     }
   }
+
+  static async commentBlogHandler(
+    req: AuthRequest,
+    res: Response) {
+    try {
+      const comment = req.body.comment;
+
+      if (!comment) {
+        res.status(400).json({ message: "Comment is required" });
+        return;
+      }
+
+      const user = await userRepository.findOneBy({ id: req.user.userId });
+      if (!user) {
+        res.status(400).json({ message: "User with this id does not exist" });
+        return;
+      }
+      const blogId = req.params.id;
+      const blog = await BlogService.getById(blogId);
+      if (!blog) {
+        res.status(400).json({ message: "Blog not found" });
+        return;
+      }
+
+      if (blog.user.id !== user.id && user.role === "user") {
```

Comment:
why this condition necessary ?

Reply:
until we create role based Auth , I think it's good to check user role while modifying record. in line 149, if condition checks it's comment owner and not admin...